backend/routes/auth.js

```diff
@@ -0,0 +1,78 @@
+const router = require('express').Router()
+const bcrypt = require('bcrypt')
+const {PrismaClient} = require('@prisma/client')
+
+const prisma = new PrismaClient();
+
+router.post('/signup', async(req, res) => {
+    try {
+        const {username, password} = req.body
+
+    if (!username || !password) { // purposefully ambigious
+        return res.status(400).json({error: "Username and password are required"})
+    }
+
+    if (password.length < 8) {
+        return res.status(400).json({error: "password must be at least 8 characters long"})
+    }
+
+    const exisitingUser = await prisma.user.findUnique({
+        where: {username}
+    })
+
+    if (exisitingUser) {
+        // if there is an existing user
+        return res.status(400).json({error: "User already exists"})
+    }
+
+    const hashedPassword = await bcrypt.hash(password, 10); // where 10 is the number of times the password is hashed
+
+    const newUser = await prisma.user.create({
+        data: {
+            username,
+            password:hashedPassword,
+        }
+    })
+
+    res.status(201).json({message: "Sign in Successful!"})
```

Comment:
we're probably gonna need to return the user id to client here so that it can be used in other requests right? (only if we want the sign up functionality to automatically login as well)